pip/_vendor/README.rst

```diff
@@ -2,11 +2,84 @@ Policy
+``pip/_vendor/vendor.txt``. In addition, any modifications must be noted in
+``pip/_vendor/README.rst``.
+
+
+Rationale
+---------
+
+Historically pip has not had any dependencies except for setuptools itself,
+choosing instead to implement any functionality it needed to prevent needing
+a dependency. However, starting with pip 1.5 we begun to replace code that was
+implemented inside of pip with reusable libraries from PyPI. This brought the
+typical benefits of reusing libraries instead of reinventing the wheel like
+higher quality and more battle tested code, centralization of bug fixes
+(particularly security sensitive ones), and better/more features for less work.
+
+However, there is several issues with having dependencies in the traditional
+way (via ``install_requires``) for pip. These issues are:
+
+* Fragility. When pip depends on another library to function then if for
+  whatever reason that library either isn't installed or an incompatible
+  version is installed then pip ceases to function. This is of course true for
+  all Python applications, however for every application *except* for pip the
+  way you fix it is by re-running pip. Obviously when pip can't run you can't
+  use pip to fix pip so you're left having to manually resolve dependencies and
+  installing them by hand.
+
+* Making other libraries uninstallable. One of pip's current dependencies is
+  the requests library, for which pip requires a fairly recent version to run.
+  If pip dependended on requests in the traditional manner then we'd end up
+  needing to either maintain compatibility with every version of requests that
+  has ever existed (and will ever exist) or some subset of the versions of
+  requests available will simply become uninstallable depending on what version
+  of pip you're using. This is again a problem that is technically true for all
+  Python applications, however the nature of pip is that you're likely to have
+  pip installed in every single environment since it is installed by default
+  in Python, in pyvenv, and in virtualenv.
+
+* Security. On the surface this is oxymoronic since traditionally vendoring
+  tends to make it harder to update a dependent library for security updates
+  and that holds true for pip. However given the *other* reasons that exist for
+  pip to avoid dependencies the alternative (and what was done historically) is
+  for pip to reinvent the wheel itself. This led to pip having implemented
+  its own HTTPS verification routines to work around the lack of ssl
+  validation in the Python standard library which ended up having similar bugs
+  to validation routine in requests/urllib3 but which had to be discovered and
+  fixed independently. By reusing the libraries, even though we're vendoring,
+  we make it easier to keep pip secure by relying on the great work of our
+  dependencies *and* making it easier to actually fix security issues by simply
+  pulling in a newer version of the dependencies.
+
```

Comment:
Also, bootstrapping? In order to install pip, you'd need to pull in any non-vendored dependencies, and if you don't yet have a functioning pip because of missing dependencies, you can't do that.